metaflow/plugins/cards/card_client.py

```diff
@@ -0,0 +1,191 @@
+from metaflow.datastore import DATASTORES, FlowDataStore
+from .card_resolver import resolve_paths_from_task, resumed_info
+from .card_datastore import CardDatastore
+from .exception import UnresolvableDatastoreException
+import os
+import tempfile
+
+
+class Card:
+    def __init__(
+        self,
+        card_ds,
+        type,
+        path,
+        html=None,
+        created_on=None,
+        from_resumed=False,
+        origin_pathspec=None,
+    ):
+        # private attributes
+        self._path = path
+        self._html = html
+        self._created_on = created_on
+        self._card_ds = card_ds
+
+        # public attributes
+        self.type = type
+        self.from_resumed = from_resumed
+        self.origin_pathspec = origin_pathspec
+
+        # Tempfile to open stuff in browser
+        self._temp_file = None
+
+    @property
+    def html(self):
+        if self._html is not None:
+            return self._html
+        self._html = self._card_ds.get_card_html(self.path)
+        return self._html
+
+    @property
+    def path(self):
+        return self._path
+
+    def browser(self):
+        import webbrowser
+
+        self._temp_file = tempfile.NamedTemporaryFile(suffix=".html")
+        html = self.html
+        self._temp_file.write(html.encode())
+        self._temp_file.seek(0)
+        url = "file://" + os.path.abspath(self._temp_file.name)
+        webbrowser.open(url)
+
+    def nb(self):
```

Comment:
don't we need `_repr_html_` in this class too, so I can do
`get_cards('foo')[0]` and see the card in a cell?